packages/markdown/loader.js

```diff
@@ -142,6 +143,47 @@ module.exports = async function demoLoader() {
     ),
   );
 
+  /**
+   * @param {*} demoName
+   * @param {*} moduleFilepath
+   * @param {*} variant
+   * @param {*} importModuleID
+   * @example detectRelativeImports('ComboBox.js', '', JS', './top100Films') => relativeModules.set('ComboBox.js', new Map([['./top100Films.js', ['JS']]]))
+   */
+  function detectRelativeImports(demoName, moduleFilepath, variant, importModuleID) {
+    if (importModuleID.startsWith('.')) {
+      let relativeModuleFilename = importModuleID;
+      const demoMap = relativeModules.get(demoName);
+      // If the moduleID does not end with an extension, we resolve it
+      // Check for an extension (ending with a "." followed by 2-4 letters)
+      if (!/\.\w{2,4}$/.test(importModuleID)) {
+        // If the demo is a JS demo, we can assume that the relative import is either
+        // a `.js` or a `.jsx` file, with `.js` taking precedence over `.jsx`
+        // likewise for TS demos, with `.ts` taking precedence over `.tsx`
+        const extensions = variant === 'JS' ? ['.js', '.jsx'] : ['.ts', '.tsx', '.js', '.jsx'];
+        const extension = extensions.find((ext) => {
+          try {
+            return statSync(path.join(moduleFilepath, '..', `${importModuleID}${ext}`));
+          } catch (error) {
+            // If the file does not exist, we return false and continue to the next extension
+            return false;
+          }
+        });
+        relativeModuleFilename = `${importModuleID}${extension}`;
```

Comment:
Maybe throw an error if you do not find the file. to say to the user 

> Hey in the dem xxx you import yyy which does not exist. PLeas ad either
> - yyy.js
> - yyy.jsx

Reply:
I thaugh this comment was nitpicking, but in https://github.com/mui/mui-x/blob/master/docs/data/date-pickers/date-picker/CustomizationExamplesNoSnap.js

The .js file imports a .tsx (which is weird) but then the error message on build time is:

> Error: Could not find a module for the relative import "./examplesConfig.stylingundefined" in the demo "CustomizationExamplesNoSnap.js"

Reply:
Yep, I saw that and the latest commit should fix this